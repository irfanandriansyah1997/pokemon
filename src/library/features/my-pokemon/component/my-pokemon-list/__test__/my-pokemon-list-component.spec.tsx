import { MockedProvider } from '@apollo/client/testing';
import { mount } from 'enzyme';
import { act } from 'react-dom/test-utils';

import { MOCK_GRAPHQL_RESPONSE } from '@/library/__mock__/graphql-response.mock';
import MyPokemonProvider from '@/library/features/my-pokemon';
import MyPokemonList from '@/library/features/my-pokemon/component/my-pokemon-list';
import SamplePokemon from '@/library/features/my-pokemon/helper/__test__/__fixture__/sample-pokemon.fixture.json';
import PokemonDialog from '@/library/features/pokemon-detail/component/pokemon-dialog';
import PokemonCard from '@/library/features/pokemon-list/component/pokemon-card';

describe(`Testing My Pokemon List Component`, () => {
  describe(`Testing Snapshot Component`, () => {
    it(`Testing With Localstorage Not Empty`, async () => {
      localStorage.setItem(`pokemon-list`, JSON.stringify(SamplePokemon));

      const ui = mount(
        <MockedProvider mocks={MOCK_GRAPHQL_RESPONSE}>
          <MyPokemonProvider>
            <MyPokemonList />
          </MyPokemonProvider>
        </MockedProvider>
      );

      expect(ui.html()).toMatchSnapshot();
      expect(ui.find(PokemonCard).length).toBe(2);
    });

    it(`Testing With Localstorage Empty`, async () => {
      localStorage.setItem(`pokemon-list`, JSON.stringify([]));

      const ui = mount(
        <MockedProvider mocks={MOCK_GRAPHQL_RESPONSE}>
          <MyPokemonProvider>
            <MyPokemonList />
          </MyPokemonProvider>
        </MockedProvider>
      );

      expect(ui.html()).toMatchSnapshot();
      expect(ui.find(PokemonCard).length).toBe(0);
    });
  });

  it(`Testing Click One of PokemonCard`, async () => {
    localStorage.setItem(`pokemon-list`, JSON.stringify(SamplePokemon));

    const ui = mount(
      <MockedProvider mocks={MOCK_GRAPHQL_RESPONSE}>
        <MyPokemonProvider>
          <MyPokemonList />
        </MyPokemonProvider>
      </MockedProvider>
    );

    expect(ui.find(PokemonCard).length).toBe(2);
    ui.find(PokemonCard).at(0).simulate(`click`);

    await act(
      () =>
        new Promise<void>((resolve) => {
          setTimeout(() => {
            ui.update();
            resolve();
          }, 1000);
        })
    );

    let showDialogProps = ui.find(PokemonDialog).at(0).prop(`showDialog`);
    const onProps = ui.find(PokemonDialog).at(0).prop(`on`);
    expect(showDialogProps).toBeTruthy();
    onProps?.({
      event: `on-close`
    });

    await act(
      () =>
        new Promise<void>((resolve) => {
          setTimeout(() => {
            ui.update();
            resolve();
          }, 1000);
        })
    );

    showDialogProps = ui.find(PokemonDialog).at(0).prop(`showDialog`);
    expect(showDialogProps).toBeFalsy();
  });
});
