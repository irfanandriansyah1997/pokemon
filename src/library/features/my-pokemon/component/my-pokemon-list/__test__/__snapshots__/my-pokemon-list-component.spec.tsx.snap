// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing My Pokemon List Component Testing Snapshot Component Testing With Localstorage Empty 1`] = `"<div class=\\"css-1m5be6d\\"><div style=\\"display: flex; flex-direction: column; margin: auto; max-width: 500px; width: 100%;\\"><div class=\\"css-18ws7yw\\"><p color=\\"primary\\" font-size=\\"medium\\" font-weight=\\"500\\" class=\\"css-g3tsjf\\">Pokemon</p><p font-size=\\"text\\" class=\\"css-1brd6hq\\">You don\`t have any pokemon</p></div></div><div color=\\"#ddd\\" class=\\"css-1b90bes\\"></div><div class=\\"css-1b90bes\\"></div><button class=\\"css-j4r94v\\"><div class=\\"poke-ball css-adl4sn\\"></div></button><div color=\\"rgba(0,0,0,0.25)\\" class=\\"css-133wia2\\"></div><div class=\\"css-swco5x\\"><div class=\\"css-ulx48j\\"><div class=\\"css-8prdx5\\"><p font-size=\\"normal\\" color=\\"primary\\" font-weight=\\"500\\" class=\\"css-nftjgg\\">Catch Pokemon Succeess!</p><div class=\\"css-136ng3n\\"><p font-size=\\"medium\\" class=\\"css-1109ac7\\">×</p></div></div><div class=\\"css-we45xm\\"><input type=\\"text\\" placeholder=\\"Give a pokemon name\\" class=\\"css-5e2eqg\\" value=\\"\\"><p color=\\"red\\" class=\\"css-1gkutas\\"></p></div><div class=\\"css-1m7yzt4\\"><button type=\\"button\\" disabled=\\"\\" class=\\"css-jsg0f6\\">Save</button><button type=\\"button\\" class=\\"css-bbsgag\\">Cancel</button></div></div></div><div color=\\"rgba(0,0,0,0.25)\\" class=\\"css-133wia2\\"></div><div class=\\"css-swco5x\\"><div class=\\"css-ulx48j\\"><div class=\\"css-8prdx5\\"><p font-size=\\"normal\\" color=\\"primary\\" font-weight=\\"500\\" class=\\"css-nftjgg\\">Failed</p><div class=\\"css-136ng3n\\"><p font-size=\\"medium\\" class=\\"css-1109ac7\\">×</p></div></div><div class=\\"css-we45xm\\"><p class=\\"css-5jbmio\\">It ran away...</p></div></div></div><div color=\\"rgba(0,0,0,0.25)\\" class=\\"css-133wia2\\"></div><div class=\\"css-swco5x\\"><div class=\\"css-ulx48j\\"><div class=\\"css-8prdx5\\"><p font-size=\\"normal\\" color=\\"primary\\" font-weight=\\"500\\" class=\\"css-nftjgg\\">Failed</p><div class=\\"css-136ng3n\\"><p font-size=\\"medium\\" class=\\"css-1109ac7\\">×</p></div></div><div class=\\"css-we45xm\\"><p class=\\"css-5jbmio\\">Are you sure delete pokemon <b style=\\"color: rgb(26, 26, 26); font-weight: 500;\\"></b> ?</p></div><div class=\\"css-1m7yzt4\\"><button type=\\"button\\" class=\\"css-1q06zjp\\">Cancel</button><button type=\\"button\\" class=\\"css-bbsgag\\">Delete</button></div></div></div></div>"`;

exports[`Testing My Pokemon List Component Testing Snapshot Component Testing With Localstorage Not Empty 1`] = `"<div class=\\"css-1m5be6d\\"><div style=\\"display: flex; flex-direction: column; margin: auto; max-width: 500px; width: 100%;\\"><div tabindex=\\"0\\" color=\\"#ddd\\" role=\\"button\\" class=\\"css-emjs67\\"><div><img loading=\\"lazy\\" alt=\\"Pokemon ivysaur\\" src=\\"https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/2.png\\"><div class=\\"pokemon-text\\"><p font-size=\\"large\\" font-weight=\\"700\\" color=\\"white\\" class=\\"css-z9ljcq\\">#002</p><p font-size=\\"normal\\" font-weight=\\"400\\" color=\\"white\\" class=\\"css-1r2ireq\\">sample-testing</p></div><div class=\\"poke-ball css-adl4sn\\"></div></div></div><div tabindex=\\"0\\" color=\\"#ddd\\" role=\\"button\\" class=\\"css-emjs67\\"><div><img loading=\\"lazy\\" alt=\\"Pokemon wartortle\\" src=\\"https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/8.png\\"><div class=\\"pokemon-text\\"><p font-size=\\"large\\" font-weight=\\"700\\" color=\\"white\\" class=\\"css-z9ljcq\\">#008</p><p font-size=\\"normal\\" font-weight=\\"400\\" color=\\"white\\" class=\\"css-1r2ireq\\">my-turtle</p></div><div class=\\"poke-ball css-adl4sn\\"></div></div></div></div></div>"`;
